src/main/java/duke/command/Parser.java

```diff
@@ -0,0 +1,91 @@
+package duke.command;
+
+import duke.utilities.DukeException;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Parser to make sense of user input.
+ */
+public class Parser {
+    /**
+     * Parses user input to a standard format
+     * @param input The user input
+     * @return The standardised input.
+     * @throws DukeException Handles duke related exceptions.
+     */
+    public static String parseCommand(String input) throws DukeException {
+
+        String[] segments = input.split(" ");
+
+        if (input.equals("list")) {
+            return "list|";
+        } else if (segments[0].equals("find")) {
+            input = input.replace("find", "");
+            if (input.equals("")) {
+                throw new DukeException("The description of a todo cannot be empty.");
+            }
+            return "find|" + input.trim();
+        } else if (segments[0].equals("mark")) {
+            int taskId;
+            if (segments[1].equals("")) {
+                taskId = -1;
+            } else {
+                taskId = Integer.parseInt(segments[1]);
+            }
+            return "mark|" + taskId;
+        } else if (segments[0].equals("unmark")) {
+            int taskId;
+            if (segments[1].equals("")) {
+                taskId = -1;
+            } else {
+                taskId = Integer.parseInt(segments[1]);
+            }
+            return "unmark|" + taskId;
+        } else if (segments[0].equals("todo")) {
```

Comment:
Perhaps it may be better to abstract out some of the logic so that the function is not too long? One rule of thumb i read is to keep function length to be a screen size at max